server/pulp/server/webservices/views/util.py

```diff
@@ -61,7 +82,8 @@ def wrapper(*args, **kwargs):
             return func(*args, **kwargs)
 
         try:
-            request.body_as_json = json.loads(request.body)
+            request_json = json.loads(request.body)
+            request.body_as_json = _ensure_input_encoding(request_json)
```

Comment:
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

On 02/18/2015 11:33 AM, Austin wrote:
> Can you move this line to an |else|? We are trying to have only one
> line in a |try| so it is clear what caused the exception.

I'm not sure I agree with being so strict about this. The point of the
try/Except block is to handle exceptional situations. If the handling
needs to be done the same for a group of lines that could all raise
the same Exception, the DRY principle says you should group them. For
example, suppose one(), two(), and three() might all raise a
ValueError, and suppose that if any one of them do this, I need to
handle it the same way. This proposed policy would mean I'd have to
write this:

```
try:
    one()
Except ValueError:
    remote_connection.close()
    _logger.info("Closed remote connection")
    local_file.close()
try:
    two()
Except ValueError:
    remote_connection.close()
    _logger.info("Closed remote connection")
    local_file.close()
try:
    three()
Except ValueError:
    remote_connection.close()
    _logger.info("Closed remote connection")
    local_file.close()
```

I think this is much cleaner, and you really don't lose anything that
is important:

```
try:
    one()
    two()
    three()
Except ValueError:
    remote_connection.close()
    _logger.info("Closed remote connection")
    local_file.close()
```

My point is not that we should always do this latter pattern, it's
that we shouldn't be so quick to make strict rules about it that don't
have any clear benefit. I think we should do what makes the most sense
in each individual situation.
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1

iQIcBAEBAgAGBQJU5M2aAAoJEIyFaKUJtmpiarYP/RBMsfNTrnaje4v/5J/B7oeT
wl9H8tDSZ0ELGaNx3dZjiJhkUzPfCsX22/6AFNBk1n+U0WPebPKJ1/8+YfXehTrM
LuC8Vxn1Zm8rhT0puVmHh6UzvWP7l10ucca8F4g+wbm1yhQth0dIQkRue7ByOvE9
EvUyU3k8pmfaLGfESUSSIxnD8AjyR3mosev2DfKrJXcrW3LiS9AJD7+nO8GitRbS
hpDTfM++1hcODriHskUplkZ4GHwETha37EShCVv+oMvfeVW3LX1XaJaPa3HdewXB
FmQTYuhBhI+nSzek1vCvhOzhNSiNidYcpnlLmr1VQink3wHk9YWyAhVEwRqjYqas
juBGITZtzFxGp30cKVR3lOehrz1KEbClVisiAPCBM67ZmVqwOmfm2/FMPm4aLala
xLIS+Oq4vTSokaNSyV5YVvApn8A+mpO8/Q2l+PEZuaNzTtCS9tx2+2kC3S9QIOwm
Gz6iXIlyc07lYvFTA23Rj4jsQ+Kj+NGXkWEQ2T5nxjpGBjhz7xV4YWpUmV3hvm/d
ialFHYTpoYJ9Q1vOBXaU/EOHnUJ7C2FVBWQRh62BX0PVvXgHpOPZvCdg+imNe3ig
AIg2ZqmZPp2npgU02mwBUQOQ0/6AjFXigyx9RlNVZkJwI8x7lnkWASdkczsKvAth
iSZgWWZfDGwTZ1BdaUKM
=iKTz
-----END PGP SIGNATURE-----